tests/test_config.py

```diff
@@ -93,100 +96,114 @@ def test_config_sphinx_command(cli, temp_with_override, file_regression):
         line
         for line in result.output.splitlines()
-        if not line.startswith("globaltoc_path")
+        if not line.startswith("external_toc_path")
     )
     file_regression.check(output, encoding="utf8")
 
 
-@pytest.mark.parametrize(
-    "toc_file, filename",
-    [("p.md", "p.md"), ("p", "p.md"), ("[]p", "[]p.md"), ("[t]p.md", "[t]p.md")],
-)
-def test_only_build_toc_files(testdir, toc_file, filename):
-    cli_config = {"latex_individualpages": False}
-    toc = Path("toc.yml")
-    toc.write_text(f"- file: '{toc_file}'\n")
-    Path(filename).write_text("")
-    Path("exclude.md").write_text("")
-    user_config = {"only_build_toc_files": True}
-
-    final_config, metadata = get_final_config(
-        toc, user_config, cli_config, validate=True, raise_on_invalid=True
-    )
-
-    assert "exclude.md" in final_config["exclude_patterns"]
-    assert filename not in final_config["exclude_patterns"]
-
-
-def test_only_build_toc_files_with_exclude_patterns(testdir):
-    cli_config = {"latex_individualpages": False}
-    toc = Path("toc.yml")
-    toc.write_text("- file: landing\n")
-    Path("landing.md").write_text("")
-    Path("exclude.md").write_text("")
-    user_config = {
-        "only_build_toc_files": True,
-        "exclude_patterns": ["my/*", "patterns"],
-    }
-
-    final_config, metadata = get_final_config(
-        toc, user_config, cli_config, validate=True, raise_on_invalid=True
-    )
-
-    assert "exclude.md" in final_config["exclude_patterns"]
-    assert "my/*" in final_config["exclude_patterns"]
-    assert "patterns" in final_config["exclude_patterns"]
-    assert "landing.md" not in final_config["exclude_patterns"]
-
-
-def test_only_build_toc_files_non_default_source_dir(testdir):
-    cli_config = {"latex_individualpages": False}
-    toc = Path("toc.yml")
-    toc.write_text("- file: landing\n")
-    sourcedir = Path("s")
-    subdir = sourcedir / "subdir"
-    subdir.mkdir(parents=True)
-    Path(sourcedir / "landing.md").write_text("")
-    Path(sourcedir / "exclude.md").write_text("")
-    Path(subdir / "sub.md").write_text("")
-    user_config = {"only_build_toc_files": True}
-
-    final_config, metadata = get_final_config(
-        toc,
-        user_config,
-        cli_config,
-        validate=True,
-        raise_on_invalid=True,
-        sourcedir=sourcedir,
-    )
-
-    assert "exclude.md" in final_config["exclude_patterns"]
-    assert "subdir/sub.md" in final_config["exclude_patterns"]
-    assert "landing.md" not in final_config["exclude_patterns"]
-
-
-def test_only_build_toc_files_missing_toc(testdir):
-    cli_config = {"latex_individualpages": False}
-    user_config = {"only_build_toc_files": True}
-
-    with pytest.raises(ValueError, match=r".*you must have a toc.*"):
-        get_final_config(
-            None, user_config, cli_config, validate=True, raise_on_invalid=True
-        )
+# TODO sphinx-external-toc now handles appending to exclude_patterns
```

Comment:
In general I am a fan of adding tests here that are *specific* to Jupyter Book functionality, but delegating testing of the logic that `sphinx-external-toc` handles to that package, so that we're not duplicating lots of complex testing infra.

Reply:
note `sphinx-external-toc` is actually very good to use to generate parts of the testing infrastucture,
i.e. you can use the `create_site_from_toc` function to write a whole site folder/file structure into a temp folder from only a toc file, then run jupyter-book against that, rather than having to store whole site folder/file structures within the tests folder

see e.g. https://github.com/executablebooks/sphinx-external-toc/blob/b17f9555094070788ae86fdd7b5f95fadcb5d13b/tests/test_sphinx.py#L59